<a href="https://colab.research.google.com/github/Musadiq8699/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Musadiq8699/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My Lane: Lane 2 (Refresh / Content Opportunity Scoring)
I chose this lane because for any organization, it is critical to identify which high-value content is losing visibility and needs to be updated to remain competitive. By prioritizing pages experiencing traffic decay, we can help the editorial team focus their limited resources on the content that will yield the highest return on investment.

In [31]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

import pandas as pd
df = pd.read_csv('flyrank-ml-internship/data/raw/content_refresh_anonymized.csv')

# Decay rate showing what percentage of the total pages are getting decayed
down_trend = df[df['trend_direction'] == 'down']
total_pages = df.shape[0]
down_pages = down_trend.shape[0]
decay_rate = (down_pages / total_pages) * 100
print(decay_rate)

# visibility impact shows how much total impressions are at stake if pages are not updated rightfully
visibility_impact = down_trend['impressions_90d'].sum()
print(visibility_impact)

#average lifespan for the decaying page
average_lifespan = down_trend['content_age_days'].mean()
print(average_lifespan)

54.20666666666667
79994363
236.17863731398353


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

**The Question:** Which of our high-performing content items are exhibiting significant traffic decay, and how should they be prioritized for a refresh? Our unit of analysis is a unique pseudonymized page (content_hash_id).  **The Numbers Behind the Problem:**Our initial data audit reveals that a staggering **54.2%** of all tracked pages are currently on a downward trend.  This decline directly threatens a massive **79,994,363** search impressions over a 90-day window.  The average age of these decaying pages sits at **236.2 days**, indicating a critical content freshness threshold where performance begins to drop.  **The Decision & Action:** This model improves the decision of where to allocate limited editorial resources first. Instead of editors manually auditing thousands of old pages, they will act on our top-ranked recommendations queue, systematically refreshing high-opportunity pages to reclaim search visibility.
**The Cost of a Wrong Call:**
False Positive
(Flagging a healthy page): The cost is purely human effort—editors spend valuable hours updating and modifying a page that did not actually require a refresh.  False Negative (Missing a decaying page): The cost is high and cumulative—high-volume pages continue their traffic slide undetected, resulting in a permanent loss of search visibility and organic traffic to competitors.

In [32]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

df.groupby('trend_direction')[['avg_position', 'ctr']].mean()


,avg_position,ctr
trend_direction,,
down,15.936305,0.324138
flat,11.102431,1.376753
new,9.912433,1.296521
stable,16.332623,0.517321
up,22.512739,0.565533


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

To prove that Lane 2 is highly viable and worth pursuing over the next 7 weeks, we loaded the starter dataset and extracted key metrics:  
**Metric 1 (High Decay Volume):** A staggering 54.2% of the entire content catalog is currently in a "down" trend, directly threatening 79,994,363 search impressions over a 90-day period.

**Metric 2 (The CTR Deficit):** Pages in a "down" trend have a severely depressed average Click-Through Rate of 0.32%, compared to 0.52% for stable pages and 0.57% for pages that are trending up.

**Metric 3 (The Decay Timeline):** The average age of decaying content sits at 236.2 days, indicating a clear "freshness cliff" where older content begins to sharply lose its performance edge.

In [33]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 4. Careful words: what I can and can't claim

**What we CAN claim (Observed, Directional, Decision-Support):**

We can identify observational and directional associations between certain page features **(like content age, low CTR, and average search position)** and the likelihood of traffic decline.  We can build a decision-support model to prioritize which pages have the highest risk and impact, allowing teams to optimize their review queue.  
**What we CANNOT claim (Causal proof, "Predicting Google"):**

We cannot claim that our model proves how Google's ranking algorithm works.  We cannot guarantee that performing a content refresh will definitely cause a recovery in traffic (proving causal impact would require an active A/B experiment or causal design, not just historical data).

In [34]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.